src/Microsoft.DotNet.ProjectModel/GlobalSettings.cs

```diff
@@ -73,6 +58,41 @@ public static bool TryGetGlobalSettings(string path, out GlobalSettings globalSe
             return true;
         }
 
+        public static GlobalSettings GetGlobalSettings(Stream fs, string globalJsonPath)
+        {
+            var globalSettings = new GlobalSettings();
+
+            var reader = new StreamReader(fs);
+            JObject jobject;
+            try
+            {
+                jobject = JObject.Parse(reader.ReadToEnd());
+            }
+            catch (JsonReaderException)
+            {
+                throw new InvalidOperationException("The JSON file can't be deserialized to a JSON object.");
+            }
+
+            var projectSearchPaths = new string[] { };
+            JToken projectSearchPathsToken;
+            if (jobject.TryGetValue("projects", out projectSearchPathsToken) &&
+                projectSearchPathsToken.Type == JTokenType.Array)
+            {
+                projectSearchPaths = projectSearchPathsToken.ToObject<string[]>();
+            }
+            else if (jobject.TryGetValue("sources", out projectSearchPathsToken) &&
+                     projectSearchPathsToken.Type == JTokenType.Array)
+            {
+                projectSearchPaths = projectSearchPathsToken.ToObject<string[]>();
```

Comment:
ToObject for primitive values (string, int, etc) is fast but for complex objects like an array it will call the serializer.